wire-ios/Wire-iOS/Sources/Helpers/syncengine/Conversation+Participants.swift

```diff
@@ -38,66 +39,110 @@ extension ZMConversation {
     }
 
     func addOrShowError(participants: [UserType]) {
-        guard let session = ZMUserSession.shared(),
-                session.networkState != .offline else {
-            self.showAlertForAdding(for: NetworkError.offline)
-            return
+        guard
```

Comment:
Where is this called in the UI? Can we modify it to pass the userSession?
I would suggest to move all this to a useCase. UI calls the UseCase.

Reply:
Is this what you had in mind ? https://github.com/wireapp/wire-ios/pull/696/commits/237afb77d602d4f9abfec340f21df4498aa5b2ba

Reply:
If we agree the implementation is right then I'll do the same for removing participants and I'll add tests

Reply:
will be done here https://wearezeta.atlassian.net/browse/WPB-5137